Update all deps

carles-grafana opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/grafana/xk6-client-tracing?pullRequest=138) <br/>All committers have signed the CLA.

carles-grafana committed:
Update all deps

- Migrate from goja to sobek (k6's fork)
- Update OpenTelemetry Collector and API types
- Replace deprecated jaegerexporter with OTLP (Jaeger supports OTLP natively)
- Update Dockerfile to use Go 1.25.4

carles-grafana committed:
Add support for custom resource attributes

Allow setting custom resource attributes (e.g., k8s.pod.uid, k8s.namespace.name)
on span templates via the new ResourceAttributes field.
